### PR TITLE
feature/Nested list 

### PR DESCRIPTION
**Pull request by eladex**

### CODE REVIEW

File: src/api/groups/api.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+//   if (depth === 0 || parent.isAleaf) {
+//     // remove 'children' array (useful for redux state)
+//     return [removeChildren(parent)];
+//   }
+//   // fetch children recursively, flatten results
+//   const offspring =  _.flatten(await Promise.all(parent.children.map(id => fetchChildren(id, depth - 1))));
+  
+//   // return the parent with it's offspring
+//   return [parent].concat(offspring);
+// }
```
Comment (eladex) [inline src/api/groups/api.js:55]: delete

File: src/components/TreeListDemo/wrapper.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const newprops = {...props, renderDummy, key, nodeId: key}
+  return React.createElement(Component, newprops);
+  // return (
+  //   <Component {...newprops} nodeId={key} key={key}/>
+  // );
+}
+
+export const wrapNodeId = (props, Component) => {
+  
+}
```
Comment (eladex) [inline src/components/TreeListDemo/wrapper.js:30]: delete file

File: src/utils/slices/helpers.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const createIdMap = (entries, path = 'id') => {
```
Comment (eladex) [inline src/utils/slices/helpers.js:1]: add docs

File: src/utils/slices/helpers.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+  const mapArr = entries.map(e => ({ [e[path]]: e }));
+  return Object.assign(...mapArr);
+};
```
Comment (eladex) [inline src/utils/slices/helpers.js:4]: add docs

File: src/store.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+const store = configureStore({
+  reducer: rootReducer,
+  middleware: [
+    ...getDefaultMiddleware(), 
+  sagaMiddleware]
+});
+
+sagaMiddleware.run(rootSaga);
+
+export default store;
```
Comment (eladex) [inline src/store.js:17]: move to features/

File: src/features/groups/groupsSlice.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
+import { createSlice, createSelector } from '@reduxjs/toolkit';
+import { createIdMap } from 'utils/slice.helpers';
+import mockIntialState from 'api/mockInitialState';
+
+const groupsSlice = createSlice({
+  name: 'groups',
+  initialState: mockIntialState.groups,
```
Comment (eladex) [inline src/features/groups/groupsSlice.js:7]: move initial state to here

**Comment by DavidHeymann** [comment]
After too many weeks, we're finally memargegim 🚀

**Pull request merged**